src/field_theory/algebraic_closure.lean

```diff
@@ -323,8 +323,20 @@ end
 instance : is_alg_closed (algebraic_closure k) :=
 is_alg_closed.of_exists_root _ $ λ f, exists_root k
 
-instance : algebra k (algebraic_closure k) :=
-(of_step k 0).to_algebra
+instance {R : Type*} [comm_semiring R] [alg : algebra R k] :
+  algebra R (algebraic_closure k) :=
+((of_step k 0).comp (@algebra_map _ _ _ _ alg)).to_algebra
```

Comment:
I vaguely worry that this will produce a `has_scalar` instance incompatible with one from a weaker `module` structure when `module R k`. I can't see any evidence that that module structure exists at the moment though, so I guess that's not a concern yet.

Reply:
You mean issues like incompatibility with the default `has_scalar ℕ _` instance? In fact, I did multiply with `ℕ` in combination with this new algebra structure, and I saw no coherence issues that wouldn't be ordinary `is_scalar_tower` applications.

Reply:
No, I mean that if you were to declare something like
```
instance {R : Type*} [comm_semiring R] [alg : module R k] :
  module R (algebraic_closure k) := sorry  -- presumably this is true?
```
further up the file, then the algebra instance would not have a defeq smul.

Reply:
Ah I see. To be honest, I don't think it's very likely that someone will ever need a `module R (algebraic_closure k)` instance that doesn't send `1 : R` to `1 : algebraic_closure k`, so I'd like to classify this as "we'll fix it if we need it".

Reply:
Right, the instance would be propositionally equal but it's likely not be to definitionally equal.